src/test/java/com/marklogic/kafka/connect/source/DslConstraintInjectionTest.java

```diff
@@ -1,61 +1,122 @@
-        String originalDsl = "op.fromView(\"Medical\", \"Authors\")  \n    .limit(1000)";
-        Map<String, Object> parsedConfig = new HashMap<String, Object>() {{
-            put(MarkLogicSourceConfig.DSL_QUERY, originalDsl);
-            put(MarkLogicSourceConfig.CONSTRAINT_COLUMN_NAME, constraintColumn);
-        }};
-        DslQueryHandler dslQueryHandler = new DslQueryHandler(null, parsedConfig);
-        String expectedValue = "op.fromView(\"Medical\",\"Authors\").where(op.gt(op.col(\"" + constraintColumn + "\"), \"" + constraintValue + "\")).limit(1000)";
-        Assertions.assertEquals(expectedValue, dslQueryHandler.injectConstraintIntoQuery(constraintValue));
+        String userDsl = "op.fromView('Medical', 'Authors')  \n    .limit(1000)";
+        String expectedValue = "op.fromView('Medical','Authors').where(op.gt(op.col('ID'), '2')).limit(1000)";
+        assertEquals(expectedValue, injectValueIntoQuery(userDsl, parsedConfig, "2"));
     }
 
     @Test
     void testQueryWithSpacesInSingleQuotes() {
-        String originalDsl = "op.fromView(\"Medical\", \"Authors\").select([op.as(\"Last Name\", op.col(\"LastName\"), op.as(\'Fore Name\', op.col(\'ForeName\')])";
-        Map<String, Object> parsedConfig = new HashMap<String, Object>() {{
-            put(MarkLogicSourceConfig.DSL_QUERY, originalDsl);
-            put(MarkLogicSourceConfig.CONSTRAINT_COLUMN_NAME, constraintColumn);
-        }};
-        DslQueryHandler dslQueryHandler = new DslQueryHandler(null, parsedConfig);
-        String expectedValue = "op.fromView(\"Medical\",\"Authors\").where(op.gt(op.col(\"" + constraintColumn + "\"), \"" +
-            constraintValue + "\")).select([op.as(\"Last Name\",op.col(\"LastName\"),op.as(\'Fore Name\',op.col(\'ForeName\')])";
-        Assertions.assertEquals(expectedValue, dslQueryHandler.injectConstraintIntoQuery(constraintValue));
+        String userDsl = "op.fromView('Medical', 'Authors').select([op.as('Last Name', op.col('LastName'), op.as('Fore Name', op.col('ForeName')])";
+        String expectedValue = "op.fromView('Medical','Authors').where(op.gt(op.col('ID'), '2')).select([op.as('Last Name',op.col('LastName'),op.as('Fore Name',op.col('ForeName')])";
+        assertEquals(expectedValue, injectValueIntoQuery(userDsl, parsedConfig, "2"));
+    }
+
+    @Test
+    void wordQueryWithPhraseOrderByAndLimit() {
+        String userDsl = "op.fromDocUris(cts.wordQuery('my phrase'))";
+        String expectedResult = userDsl + ".where(op.gt(op.col('ID'), '2')).orderBy(op.asc('ID')).limit(1000)";
+        assertEquals(expectedResult, injectValueIntoQuery(userDsl + ".orderBy(op.asc('ID')).limit(1000)", parsedConfig, "2"),
+            "The where clause should have been injected just after the closing paren of the fromDocUris function");
+    }
+
+    @Test
+    void wordQueryWithPhraseAndFollowOnFunction() {
+        String userDsl = "op.fromDocUris(cts.wordQuery('my phrase').followOn('abc'))";
```

Comment:
I know the query's not being executed, but just to make it realistic, swap out "followOn" for "joinDoc" because that's likely what a user would use next.